components/Navigation.tsx

```diff
@@ -65,13 +62,14 @@ const Navigation = (): JSX.Element => {
       fontSize: 14,
     },
   }));
-  // const toggleLang = () => {
-  //   if (langText === 'EN') {
-  //     router.replace(router.pathname.substring(3) || '/');
-  //   } else {
-  //     router.replace('/zh' + router.pathname);
-  //   }
-  // };
+
+  const toggleLang = () => {
+    if (langText === 'EN') {
+      router.replace(router.asPath.substring(3) || '/');
+    } else {
+      router.replace('/zh' + router.asPath);
+    }
```

Comment:
The logic here should be more complex. For example, if the visitor is visiting a detail EIP page, when they click the language button, they might hope to be redirected to the /zh/eip-xxxx page.